File: server/java/src/main/java/com/onlinepayments/example/service/dtos/additionalpaymentactions/AdditionalPaymentActionRequestDto.java
package com.onlinepayments.example.service.dtos.additionalpaymentactions;

import com.onlinepayments.example.service.domain.common.enums.Currency;

import java.math.BigDecimal;

public record AdditionalPaymentActionRequestDto(
        String id,
        BigDecimal amount,
        Currency currency,
        Boolean isFinal
) { }

File: server/dotnet/sdk-dotnet-example/Business/DTOs/AdditionalPaymentActions/AdditionalPaymentActionRequestDto.cs
using Business.Domain.Common.Enums;

namespace Business.DTOs.AdditionalPaymentActions;

public class AdditionalPaymentActionRequestDto
{
    public decimal Amount { get; set; }

    public Currency? Currency { get; set; }

    public bool? IsFinal { get; set; }

    public required string Id { get; set; }
}